File: packages/poc/tests/test_greeting.py
from octobot_poc import greet, add, fibonacci


def test_greet():
    assert greet("World") == "Hello, World!"
    assert greet("") == "Hello, !"
    assert greet("OctoBot") == "Hello, OctoBot!"


def test_add():
    assert add(2, 3) == 5
    assert add(-1, 1) == 0
    assert add(0, 0) == 0


def test_fibonacci():
    assert fibonacci(0) == 0
    assert fibonacci(1) == 1
    assert fibonacci(2) == 1
    assert fibonacci(10) == 55
    assert fibonacci(20) == 6765

File: packages/poc/crates/poc_core/tests/test_greeting.rs
use poc_core::greeting::{greet, add, fibonacci, VERSION};

#[test]
fn test_version() {
    assert_eq!(VERSION, "0.1.0");
}

#[test]
fn test_greet() {
    assert_eq!(greet("World"), "Hello, World!");
    assert_eq!(greet(""), "Hello, !");
    assert_eq!(greet("OctoBot"), "Hello, OctoBot!");
}

#[test]
fn test_add() {
    assert_eq!(add(2, 3), 5);
    assert_eq!(add(-1, 1), 0);
    assert_eq!(add(0, 0), 0);
    assert_eq!(add(i64::MAX, 0), i64::MAX);
}

#[test]
fn test_fibonacci() {
    assert_eq!(fibonacci(0), 0);
    assert_eq!(fibonacci(1), 1);
    assert_eq!(fibonacci(2), 1);
    assert_eq!(fibonacci(10), 55);
    assert_eq!(fibonacci(20), 6765);
    assert_eq!(fibonacci(50), 12586269025);
}
